tchannel/sync/client.py

```diff
@@ -94,6 +94,9 @@ def advertise(self, routers, name=None, timeout=None):
         :returns: first advertise result.
         :raises AdvertiseError: when unable to begin advertising.
         """
+        # begin listening if hasn't already
+        if not self.async_client.is_listening():
+            self.async_client.listen()
```

Comment:
Thanks for catching... I started it here and then moved it to the async client; guess I forgot to remove it from here.